test/jdk/sun/net/www/http/HttpClient/KeepAliveTest.java

```diff
@@ -0,0 +1,590 @@
+
+            /* send the header */
+            out.write("HTTP/1.1 200 OK\r\n");
+            out.write("Content-Type: text/plain; charset=iso-8859-1\r\n");
+
+            /*
+             * append each scenario content from array.
+             */
+            if(serverScenarioContent != null) {
+                out.write(serverScenarioContent);
+            }
+            out.write(CLEN);
+            out.write(NEW_LINE);
+            out.write(BODY);
+            out.flush();
+        } catch (IOException e) {
+            e.printStackTrace();
+        } finally {
+            if (out != null) {
+                out.flush();
+                out.close();
+            }
+            if (socket != null) {
+                socket.close();
+            }
+            if (serverSocket != null) {
+                serverSocket.close();
+            }
+        }
+    }
+
+    private void clientInit(int scenarioNumber) throws Exception {
+        try {
+            createURLClient(scenarioNumber);
+        } finally {
+            System.out.println("client count down latch:" + scenarioNumber);
+            this.countDownLatch.countDown();
+            System.out.println();
+            System.out.println();
+        }
+    }
+
+    private void createURLClient(int scenarioNumber) throws Exception {
+
+        //    System.setProperty("java.net.useSystemProxies", "false");
+        //    System.setProperty("http.nonProxyHosts", "");
+        //    System.setProperty("http.proxyHost", "localhost");
+        //    System.setProperty("http.proxyPort", String.valueOf(SERVER_PORT));
+        System.out.println("Following are Existing System Properties if set any");
+        System.out.println("http.keepAlive.time.server:" + System.getProperty("http.keepAlive.time.server"));
+        System.out.println("http.keepAlive.time.proxy:" + System.getProperty("http.keepAlive.time.proxy"));
+        System.setProperty("java.net.useSystemProxies", "false");
+        System.out.println("http.proxyPort:"+System.getProperty("http.proxyPort"));
+        System.out.println("http.proxyHost:"+System.getProperty("http.proxyHost"));
+        System.clearProperty("http.keepAlive.time.server");
+        System.clearProperty("http.keepAlive.time.proxy");
+
+        // fetch clientScenearios for each scenarioNumber from array and set it to
+        // System property.
+        if (!clientScenarios[scenarioNumber].equalsIgnoreCase(NI)) {
+            System.out.println("Client Input Parsing");
+            for (String clientScenarioString : clientScenarios[scenarioNumber].split(CLIENT_SEPARATOR)) {
+                System.out.println(clientScenarioString);
+                String key = clientScenarioString.split("=")[0];
+                String value = clientScenarioString.split("=")[1];
+                System.setProperty(key, value);
+            }
+        }
+
+        // wait until ServerSocket moves to listening state.
+        while (!serverReady) {
```

Comment:
Could use a cyclic barrier or a count down latch instead of sleeping here.

Reply:
Used serverCountDownLatch @dfuch thanks.